dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/task/statemachine/ITaskStateAction.java

```diff
@@ -89,7 +91,15 @@ void onRetryEvent(final IWorkflowExecutionRunnable workflowExecutionRunnable,
      */
     void onDispatchEvent(final IWorkflowExecutionRunnable workflowExecutionRunnable,
                          final ITaskExecutionRunnable taskExecutionRunnable,
-                         final TaskDispatchLifecycleEvent taskDispatchEvent);
+                         final TaskDispatchLifecycleEvent taskDispatchEvent) throws TaskFatalException;
+
+    /**
+     * Perform the necessary actions when the task in a certain state receive a {@link TaskFatalLifecycleEvent}.
+     * <p> This method is called when the task encounters an unrecoverable error (e.g., initialization failure).
```

Comment:
```suggestion
     * <p> This method is called when the task encounters catastrophic failure (e.g., initialization failure).
```

We will still retry the task instance if config retry policy.

Reply:
ok